tests/test_utility/test_abc.py

```diff
@@ -0,0 +1,45 @@
+from unittest import mock
+
+import pytest
+
+from PyQt5.QtCore import QObject, pyqtSignal
+from volumina.utility import abc
+
+
+class TestQAbc:
+    class IObject(abc.QABC):
+        @abc.abstractmethod
+        def foo(self):
+            ...
+
+    class ISignal(abc.QABC):
+        signal = abc.abstractsignal()
+
+    def test_type_error_on_undefined_abstract_method(self):
+
+        with pytest.raises(TypeError):
+            self.IObject()
+
+    def test_not_implemented_signal_raises_type_error(self):
+        with pytest.raises(TypeError):
+            self.ISignal()
+
+    def test_implemented_signal_doesnt_raise_type_error(self):
+        class MySignal(self.ISignal):
+            signal = pyqtSignal(int)
+
+        MySignal()
+
+    def test_qobject_signal(self):
```

Comment:
I might be wrong, but it seems that this just tests Qt signal system and nothing else because Python's MRO would look up "connect" from the first superclass, which is QObject.